### PR TITLE
Merge pull request #2 from Ramya10101999/main

### PR DESCRIPTION
**Pull request by Ramya10101999**
Merge pull request #1 from Ramya10101999/Gamming

### CODE REVIEW

File: GuessingGame.Tests/UnitTest1.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace GuessingGame.Tests;
 public class UnitTest1
```
Comment (zanehanasoft) [inline GuessingGame.Tests/UnitTest1.cs:7]: For production code you would name the unit test class something more meaningful for the use case. I usually name them after the class I am testing in your case it would be ServiceTests.

File: GuessingGame.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public void GetResult_found()
         INumberGenerator numberGenerator = Substitute.For<INumberGenerator>();
         numberGenerator.Generate(minRange, maxRange).Returns(guessNumber);
         Service service = new Service(numberGenerator);
-        string result=service.GetResult(minRange,maxRange,6);
-
+        service.GetResult(minRange,maxRange,6);
+        Assert.True(true);
```
Comment (zanehanasoft) [inline GuessingGame.Tests/UnitTest1.cs:20]: This Assert isn't asserting anything useful, it is checking if true == true.

File: GuessingGame.Tests/UnitTest1.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@ namespace GuessingGame.Tests;
 public class UnitTest1
 {
     [Fact]
-    public void GetResult_found()
+    public void GetResult_Win()
```
Comment (zanehanasoft) [inline GuessingGame.Tests/UnitTest1.cs:10]: Its also a good idea to test for the loss condition as well, that way you know you code is working correctly.

You might also need another test to make sure that given a specific difficulty it returns the correct minimum and maximum range, that way you can be sure your code also handles that important business case.

File: GuessingGame/Service.cs [hunk 4]
```diff
@@ -11,9 +11,9 @@ public Service(INumberGenerator numberGenerator)
         {
             _numberGenerator = numberGenerator;
         }
-
```
Comment (zanehanasoft) [inline GuessingGame/Service.cs:14]: In Production code you wouldn't name a class just Service, it doesn't clearly explain what it does, when you catch up to SOLID principles you will see that S stands for SRP or single responsibility Principle. If you name a class too generic then its difficult for other developers to know what the intention of the class is and they would end up adding more code to it, which would bloat the service too much.

File: GuessingGame/Service.cs [hunk 5]
```diff
@@ -11,9 +11,9 @@ public Service(INumberGenerator numberGenerator)
         {
             _numberGenerator = numberGenerator;
         }
-        
 
-        public int GetAmount()
+
+    public int GetAmount()
     {
     Console.WriteLine("Enter the amount of money to bet:");
```
Comment (zanehanasoft) [inline GuessingGame/Service.cs:18]: If this was production code you would usually indent this part of the code it would make it easier to read.

File: GuessingGame/Service.cs [hunk 6]
```diff
@@ -37,35 +37,35 @@ public int GetNumber()
...
+            Console.WriteLine("enter a correct string");
+                return GetLevel();
+                
+            }
 
     }
     public string GetResult(int minRange,int maxRange,int multiplier){
-                    Random random = new Random();
-                    int randomNumber = random.Next(minRange, maxRange);
+        Random random = new Random();
```
Comment (zanehanasoft) [inline GuessingGame/Service.cs:56]: you have dependency injection in this class but you aren't implementing it here.

File: GuessingGame/Service.cs [hunk 7]
```diff
@@ -37,35 +37,35 @@ public int GetNumber()
...
+        int randomNumber = random.Next(minRange, maxRange);
 
-                    int game = GetAmount() * multiplier;
+        int game = GetAmount() * multiplier;
```
Comment (zanehanasoft) [inline GuessingGame/Service.cs:59]: consider the name of this property, is game a good name for it?

**Comment by zanehanasoft** [comment]
@Ramya10101999 The code looks like its getting there. Update the dependency injection and the unit tests and then I will review again.

Also consider using the auto formatting tool in Visual studio code, it will put the tabs and spaces into your code for you to make it easier to read. I explain setting it up in this tutorial [https://highlow.atlassian.net/wiki/spaces/TS/pages/3909484600/Visual+Studio+Code](url)

**Comment by zanehanasoft** [comment]
Since I am unable to comment on the remainder of your code in this PR, I will attempt to do it here.

In the service class consider the name of the method GetLevel() reading that method I would assume that it returns a level but it actually calculates the result as well. It's a good idea to name the method to explain what it does or even better, to separate the responsibility in the code to make it easier to understand for other developers. Also it returns a value but that value isn't actually used when you call this method.

Normally in c# we put classes like Service in their own folder, for a small project like this it doesn't matter but for production level code you should do something like this to make it easier to organise your code.

Good job making the application recursive though, its a useful technique.

File: GuessingGame.Tests/UnitTest1.cs (hunk 3 above)
Comment (Ramya10101999) [reply GuessingGame.Tests/UnitTest1.cs:10]: I am not getting this comment please explain me.

Comment (zanehanasoft) [reply GuessingGame.Tests/UnitTest1.cs:10]: You have a unit test for the win condition but you should also have a unit test for the Loss condition too.

Your test will check whether the guess value matches the random number but there is another piece of important logic which involves using the correct min and maximum values for the difficulty chosen. ie If you pick easy then it should correctly pick the right maximum and minimum values, This test would make sure you didn't accidentally get 1-20 for picking easy. I wouldn't worry about this test as much, just something to think about when unit testing.
